### PR TITLE
[sankey] Fix for the graph bug after updating the buckets

### PR DESCRIPTION
**Pull request by ch-bas**
Some side effects were noticed after the integration of the last auto resize feature.
The first one is relative about a performance issue after each resize operation.
The second one is detected when you change the input of the buckets and you want to perform a new resize action.

### CODE REVIEW

File: public/kbn_sankey_vis_controller.js [hunk 1]
```diff
@@ -45,106 +45,104 @@ module.controller('KbnSankeyVisController', function ($scope, $element, $rootSco
...
     let energy = data.slices;
     div = d3.select(svgRoot);
     if (!energy.nodes.length) return;
 
     svg = div.append('svg')
-      .attr('width', width)
-      .attr('height', height + margin)
-      .append('g')
-      .attr('transform', 'translate(0, 0)');
+        .attr('width', width)
```
Comment (kaihil) [inline public/kbn_sankey_vis_controller.js]: Why did you increase indentation here (and at some places below) from one to two spaces? Isn't two spaces correct in terms of our coding conventions?

File: public/lib/agg_response.js [hunk 2]
```diff
@@ -2,48 +2,54 @@
 // _array_to_linked_list should be imported from the removed directory 'ui/agg_response/hierarchical/_array_to_linked_list'
 const { arrayToLinkedList } = require('./_array_to_linked_list');
 const { aggregate } = require('./agg_response_helper');
+require('ui/notify');
 
-module.exports = function sankeyProvider(Private, Notifier) {
-  let notify = new Notifier({
+module.exports = function sankeyProvider(Private, createNotifier) {
```
Comment (kaihil) [inline public/lib/agg_response.js]: Looks like you added the change from the other branch into this commit. This is kinda strange. Please keep different changes in different pull requests (or at least different commits).

**Comment by kaihil** [comment]
This works and looks good (despite my two comments) 👍

File: public/kbn_sankey_vis_controller.js (hunk 1 above)
Comment (ch-bas) [reply public/kbn_sankey_vis_controller.js]: This was actually a mistake made due the IDE when formatting  the code.
It will not occur in the future releases.

Comment (ch-bas) [reply public/kbn_sankey_vis_controller.js]: The indentations were modified now to follow the Kibana's conventions.
